packages/cli/src/commands/dev/dev.ts

```diff
@@ -108,42 +82,43 @@ async function endpointHealthChecker(parsedOptions: DevCommandOptions, endpointR
   }
 }
 
-function wait(ms: number) {
-  return new Promise((resolve) => setTimeout(resolve, ms));
-}
-
-function parseOptions(options: DevCommandOptions) {
-  const { origin, port, region } = options || {};
+async function tunnelHealthCheck(configTunnelUrl: string): Promise<boolean> {
+  try {
+    const res = await (
+      await fetch(configTunnelUrl + '?action=health-check', {
+        method: 'GET',
+        headers: {
+          accept: 'application/json',
+          'Content-Type': 'application/json',
+        },
+      })
+    ).json();
 
-  return {
-    ...options,
-    origin: origin || getDefaultOrigin(port),
-    dashboardUrl: options.dashboardUrl || getDefaultDashboardUrl(region),
-  };
+    return res.status === 'ok';
+  } catch (e) {
+    return false;
+  }
 }
 
-function getDefaultOrigin(port: string) {
-  return `http://${SERVER_HOST}:${port}`;
-}
+async function createTunnel(localOrigin: string, endpointRoute: string): Promise<string> {
+  const configTunnelUrl = config.getValue('tunnelUrl');
+  const storeUrl = configTunnelUrl ? new URL(configTunnelUrl) : null;
+  const originUrl = new URL(localOrigin);
 
-function getDefaultDashboardUrl(region: string) {
-  switch (region) {
-    case CloudRegionEnum.EU:
-      return DashboardUrlEnum.EU;
-    case CloudRegionEnum.STAGING:
-      return DashboardUrlEnum.STAGING;
-    case CloudRegionEnum.US:
-    default:
-      return DashboardUrlEnum.US;
+  if (storeUrl) {
+    await connectToTunnel(storeUrl, originUrl);
+    await wait(100);
+    const healthCheck = await tunnelHealthCheck(configTunnelUrl + endpointRoute);
```

Comment:
The main problem I'm seeing here, is that we are not checking the health of the tunnel here, but the health of the bridge app in the tunnel. 

If my local app will be down for some reason, and the tunnel is completely healthy this will fail here. 

Can we maybe expose some sort of a `health` attribute returned by the NtfrClient?

Reply:
Totally agree this was my first approach however i was not able to expense any data from the connection. Let me do a quick check again :)

Reply:
Done :)